python/lsst/ts/linearstage/controllers/zaber_lst.py

```diff
@@ -19,23 +19,229 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-__all__ = ["Zaber"]
+__all__ = ["Zaber", "ZaberV2"]
 
 import asyncio
+import logging
 import os
 import pty
+import types
 
 import yaml
-from lsst.ts.linearstage.mocks.mock_zaber_lst import MockSerial
+from lsst.ts import salobj
+from lsst.ts.linearstage.mocks.mock_zaber_lst import LinearStageServer, MockSerial
 from zaber import serial as zaber
 
+try:
+    from zaber_motion import Units
+    from zaber_motion.ascii import AxisType, Connection, Device
+    from zaber_motion.exceptions import (
+        CommandFailedException,
+        ConnectionFailedException,
+    )
+except ImportError:
+    pass
+
 from .stage import Stage
 
 _LOCAL_HOST = "127.0.0.1"
 _STD_TIMEOUT = 20  # standard timeout
 _ZABER_MOVEMENT_TIME = 3  # time to wait for zaber to complete movement/homing
 
 
+class ZaberV2(Stage):
+    """Implement Zaber stage with zaber-motion library.
+
+    Parameters
+    ----------
+    config : `types.Simplenamespace`
+        The controller specific schema.
+    log : `logging.Logger`
+        The log of the controller.
+    simulation_mode:
+        Is the controller in simulation mode?
+
+    Attributes
+    ----------
+    client : `None` | `tcpip.Client`
+        The tcpip client that commands the device.
+    mock_server : `None` | `LinearStageServer`
+        The mock server that's started if in simulation mode.
+    position : `None` | `float`
+        The position of the stage along the axis.
+
+    """
+
+    def __init__(
+        self, config: types.SimpleNamespace, log: logging.Logger, simulation_mode: int
+    ) -> None:
+        super().__init__(config, log, simulation_mode)
+        self.client: Connection | None = None
+        self.mock_server: LinearStageServer | None = None
+        self.position: float | None = None
+        self.device: Device | None = None
+
+    @property
+    def connected(self) -> bool:
+        """Is the client connected?"""
+        # Assume client is connected
```

Comment:
I think this comment would be better placed inside the `if`statement. I assume that the zaber library doesn't have a way to verify that a client is connected so please explicitly state that here too.